design/scheduled-update-check.md

```diff
@@ -0,0 +1,64 @@
+# Scheduled Update Checks for Online Installs
+
+KOTS (Admin Console) does not automatically check for new updates. This proposal adds the 
+ability for KOTS (Admin Console) to check for updates every X period of time for online installations.
+
+## Goals
+
+- Pull new updates automatically for KOTS (Admin Console) online installations
+
+## Non Goals
+
+- Deployment of the new updates
+- Scheduled Update Checks for Airgap Installations
+
+## Background
+
+Many customers assume that scheduled update checks are a given and are surprised when they do
+not receive updates automatically.
+
+## High-Level Design
+
+1. This proposal is only for online installations in KOTS Admin console.
+2. The default period of time (X) is configurable.
+3. The ability to disable this feature completely. It will be enabled by default.
+
+The update check will happen in a background process/thread which is triggered every X period of time.
+On each trigger, the process will get all available applications and run update checks for each app.
+The update check will automatically create new versions for each update available in an app.
+
+## Detailed Design
+
+3 new columns (`update_checker_enabled` & `update_checker_interval` & `update_checker_status`) will be 
```

Comment:
How about instead of adding new column, we use config maps for this?

Reply:
@divolgin this makes sense, but we are worried about "configmap explosion".  We are working on multiple projects that need config right now, and if each need a new config map, that feels a little bad. Ideally we'd have a global config map to use, here, but we don't. 

Agreed that this shouldn't be in the database, but I'm also not opposed to it for now. Would you be ok with this as is?

Reply:
Yep, db is ok.  We have some of this built already (specifically status and progress), which I mentioned in other comments.

Reply:
Right, my initial thoughts were to save those values in a configmap as well. But then i changed my mind due to the reasons marc mentioned above. Also https://github.com/replicatedhq/kots/pull/497#discussion_r420445806